src/main/java/com/sprint/mission/discodeit/controller/MessageController.java

```diff
@@ -1,61 +1,73 @@
 package com.sprint.mission.discodeit.controller;
 
-import com.sprint.mission.discodeit.DTO.Message.CreateMessageDTO;
-import com.sprint.mission.discodeit.DTO.Message.FindMessageDTO;
-import com.sprint.mission.discodeit.DTO.Message.UpdateMessageDTO;
-import com.sprint.mission.discodeit.entity.BinaryContent;
+import com.sprint.mission.discodeit.dto.Message.CreateMessageDTO;
+import com.sprint.mission.discodeit.dto.Message.MessageUpdateRequest;
 import com.sprint.mission.discodeit.entity.Message;
 import com.sprint.mission.discodeit.service.MessageService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
 import lombok.RequiredArgsConstructor;
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
 
 import java.util.List;
 import java.util.UUID;
 
-@RequestMapping("/message")
-@Controller
+@Tag(name = "Message", description = "Message API")
+@RequestMapping("/api/messages")
+@RestController
 @RequiredArgsConstructor
 public class MessageController {
     private final MessageService messageService;
 
-    @RequestMapping(value = "", method = RequestMethod.POST, consumes = "multipart/form-data")
-    public String createMessage(@RequestParam List<MultipartFile> multipartFiles,
-                                 CreateMessageDTO createMessageDTO) {
-        messageService.create(multipartFiles,createMessageDTO);
-        return"redirect:/messageP";
-    }
-
-    @RequestMapping(value = "", method = RequestMethod.PUT, consumes = "multipart/form-data")
-    public String updateMessage(@RequestParam List<MultipartFile> multipartFiles,
-                              UpdateMessageDTO updateMessageDTO) {
-        messageService.update(multipartFiles,updateMessageDTO);
-        return "redirect:/messageP";
+    @Operation(summary = "Message 생성")
+    @ApiResponse(responseCode = "201", description = "Public Channel이 성공적으로 생성됨")
+    @ApiResponse(responseCode = "404", description = "Channel 또는 User를 찾을 수 없음")
+    @PostMapping(consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
+    public ResponseEntity<Message> createMessage(
+            @Schema(description = "Message 생성 정보")
+            @RequestPart(name = "messageCreateRequest") CreateMessageDTO messageCreateRequest,
+            @Schema(description = "Message 첨부 파일들")
+            @RequestPart(name = "attachments",required = false) List<MultipartFile> attachments) {
```

Comment:
애플리케이션 상에서 실제 메시지를 보내보면 아래와 같은 에러가 발생하네요.

<img width="1728" height="981" alt="image" src="https://github.com/user-attachments/assets/8a82c576-67ab-4601-985d-99a62e202553" />


attachments를 required=false로 바꾸는게 좋을 것 같아요!

```suggestion
            @RequestPart(name = "attachments",required = true) List<MultipartFile> attachments) {
```